### DISCUSSION TITLE
How to update the access token when changing a users role in a RBAC setup

### DISCUSSION THREAD

**Discussion by NixBiks** [supabase/supabase#22614] [Questions]
I have a database with RBAC similar to [this](https://supabase.com/docs/guides/auth/custom-claims-and-role-based-access-control-rbac) guide. 

I have a simple question though (probably it isn't a simple answer); if I want to make a user `admin` then I'd add the user to  `user_roles` table. But the decoded jwt won't have this updated `user_role` until the user signs out and signs back in. Isn't there some way to force this?

**Comment by GaryAustin1** [comment]
You have to force a refresh of the token with refreshSession or signout/in.  There is no ability for the server to notify a client to get the new jwt until it expires.

This is the tradeoff of using the jwt for security versus using your roles table directly in RLS with joins.
